### PR TITLE
dev-libs/xmlsec: Bump to 1.2.30

### PR DESCRIPTION
**Pull request by thesamesam**
Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Sam James (sam_c) <sam@cmpct.info>

### CODE REVIEW

**Comment by thesamesam** [comment]
@Whissi please review.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools
```
Comment (thesamesam) [inline dev-libs/xmlsec/xmlsec-1.2.30.ebuild:6]: I think this can go, will check later.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+HOMEPAGE="https://www.aleksey.com/xmlsec"
+SRC_URI="https://www.aleksey.com/xmlsec/download/${PN}1-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~sparc ~x86"
+IUSE="doc gcrypt gnutls libressl nss +openssl static-libs test"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="|| ( gcrypt gnutls nss openssl )
+	gnutls? ( gcrypt )"
```
Comment (thesamesam) [inline dev-libs/xmlsec/xmlsec-1.2.30.ebuild:18]: I think this needs improvement to at least remove this strange gcrypt => gnutls, but maybe more. A lot of things just require both simultaneously.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	nss? (
+		>=dev-libs/nspr-4.4.1:=
+		>=dev-libs/nss-3.9:=
+	)
+	openssl? (
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl:0= )
+	)"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig
```
Comment (thesamesam) [inline dev-libs/xmlsec/xmlsec-1.2.30.ebuild:33]: Shouldn't this be at runtime too?

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		$(use_with gcrypt) \
+		$(use_with gnutls) \
+		$(use_with nss nspr) \
+		$(use_with nss) \
+		$(use_with openssl) \
+		--enable-mans \
+		--enable-pkgconfig
+}
+
+src_test() {
```
Comment (thesamesam) [inline dev-libs/xmlsec/xmlsec-1.2.30.ebuild:55]: Maybe could set SHELL here to workaround this [bug](https://bugs.gentoo.org/721128)? I did try something before, but it didn't work. Worth trying again.

Comment (Whissi) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:55]: Check if build system will respect SHELL.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild (hunk 3 above)
Comment (Whissi) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:33]: It's not clear for me about which line you are talking.

Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:33]: virtual/pkgconfig only in BDEPEND

Comment (Whissi) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:33]: I would be surprised if this is required at runtime. That's usually a BDEPEND-only candidate.

Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:33]: You're right.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild (hunk 2 above)
Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:18]: I think this is fine for now.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild (hunk 4 above)
Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:55]: Huh. I'd tried this or similar before. Works now.

File: dev-libs/xmlsec/xmlsec-1.2.30.ebuild (hunk 2 above)
Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.30.ebuild:18]: I think upon further investigation, it's okay, but strange.

**Pull request closed without merging**